pandas/io/formats/format.py

```diff
@@ -1520,17 +1520,26 @@ def __init__(
     def _format_strings(self) -> List[str]:
         """ we by definition have DO NOT have a TZ """
         values = self.values
+        flat_values = values.ravel()
 
-        if not isinstance(values, DatetimeIndex):
-            values = DatetimeIndex(values)
+        if not isinstance(flat_values, DatetimeIndex):
+            flat_values = DatetimeIndex(flat_values)
 
         if self.formatter is not None and callable(self.formatter):
-            return [self.formatter(x) for x in values]
+            flat_str_values = np.array([self.formatter(x) for x in flat_values])
+            fmt_values = flat_str_values.reshape(values.shape)
+        else:
+            fmt_values = flat_values._data._format_native_types(
```

Comment:
This will actually call `ravel()`/`reshape()` itself, but flattened values are needed above so easier to use that here as well.